roles/openshift_master/tasks/main.yml

```diff
@@ -93,7 +93,7 @@
     src: htpasswd.j2
     mode: 0600
     backup: yes
-  when: item.kind == 'HTPasswdPasswordIdentityProvider'
+  when: item.kind == 'HTPasswdPasswordIdentityProvider' and openshift.master.manage_htpasswd | bool
```

Comment:
I think we also need to ensure that the file exists in the case that manage_htpasswd is false, otherwise the master service will fail to startup if the file isn't present, something like this should be sufficient:
```
- name: Ensure htpasswd file exists
  copy:
    dest: "{{ item.filename }}"
    content: ""
    mode: 0600
    force: no
  when: item.kind == 'HTPasswdPasswordIdentityProvider' and openshift.master.manage_htpasswd | bool
  with_items: "{{ openshift.master.identity_providers }}"
```